main/resource-access.bs

```diff
@@ -75,6 +75,352 @@ Issue: Write Linked Data Platform section.
+it is directly associated with is also deleted and the Solid server is
+authoritative for both resources.
+
+A Solid server SHOULD sanity check ACL auxiliary resources upon creation or
+update to restrict invalid changes, such as by performing shape validation
+against authorization statements therein.
+
+#### Resource Description #### {#ar-description}
+
+Note: Consider where there are any common parameters that would be
+ubiquitous across resource descriptions that should be defined as part of the
+specification.
+
+Resource description is a general mechanism to provide descriptive metadata for
+a given resource. It MUST be supported as an auxiliary type by Solid
+servers.
+
+The Descriptive auxiliary resource directly associated with a given resource is
+discovered by the client via the `"rel=describedby"` `Link` relation in a `Link`
+header. Conversely, the resource being described by a Descriptive auxiliary
+resource is discovered by the client via the `rel="describes"` `Link` relation
+in a `Link` header.
+
+Issue: Consider whether a given Solid resource should be allowed to have
+multiple resource description auxiliary resources.
+[Related Issue](https://github.com/solid/specification/issues/173)
+
+A given Solid resource MUST NOT be directly associated with more than one
+Descriptive auxiliary resource.
+
+Issue: Determine what the default permissions should be on resource description
+auxiliary resources, or whether we should have them at all.
+[Related Issue](https://github.com/solid/specification/issues/174)
+
+To create or modify a Descriptive auxiliary resource, a given
+[acl:agent](https://github.com/solid/web-access-control-spec#describing-agents)
+MUST have
+[acl:Write](https://github.com/solid/web-access-control-spec#aclcontrol)
+privileges per the
+[ACL inheritance algorithm](https://github.com/solid/web-access-control-spec#acl-inheritance-algorithm)
+on the resource directly associated with it.
+
+To discover or read a Descriptive auxiliary resource, an
+[acl:agent](https://github.com/solid/web-access-control-spec#describing-agents)
+MUST have
+[acl:Read](https://github.com/solid/web-access-control-spec#aclcontrol)
+privileges per the
+[ACL inheritance algorithm](https://github.com/solid/web-access-control-spec#acl-inheritance-algorithm)
+on the resource directly associated with it.
+
+An Descriptive auxiliary resource MUST be deleted by the Solid server when the
+resource it is directly associated with is also deleted and the Solid server is
+authoritative for both resources.
+
+#### Shape Validation #### {#ar-shape}
+
+Shape Validation auxiliary resources as defined by (link to shape validation)
+SHOULD be supported as an auxiliary type by Solid servers.
+
+The Shape validation auxiliary resource directly associated with a given
+resource
+is discovered by the client via the `rel=http://www.w3.org/ns/solid/terms#shape`
+`Link` relation in a `Link` header. Conversely, the resource being described by
```

Comment:
What's the history behind having two descovery mechanisms for shape? (shape and describes)?

Reply:
We use this convention on all of the auxiliary resources that have a 1:1 relationship with the resource they have been associated with. Essentially its just a way to have a bi-directional association, specifically:

Resource A -> Shape Validation V
Shape Validation V -> Resource A

Note that the shape validation resource is not meant to actually link straight to the shape itself, but to an RDF document that will point to the shape(s), contain some other helpful metadata like type of shape(s), enforcement level - (warn, fail), etc.

Now that aux. resource section is wrapping I plan to start documenting that one, so there should be more detail soon.

Reply:
>Note that the shape validation resource is not meant to actually link straight to the shape itself, but to an RDF document that will point to the shape(s), contain some other helpful metadata like type of shape(s), enforcement level - (warn, fail), etc.

Why isn't the shape resource self-describing ie. including both the shape and anything else that's relevant?

The name of the property signals "shape", the resource is called "shape resource".. but when the client gets there, they are directed to another resource which is the actual shape?

Too many meta layers for my taste.

We have rel=acl, describedby,.. targeting the most essential resource. How is the metadata (eg. enforcement level) about the shape resource (ie. the target of rel=shape) is of use to the client?

Reply:
> Why isn't the shape resource self-describing ie. including both the shape and anything else that's relevant?

The shape resource is public and doesn't know anything about the context in which you are applying it. For example, you may only want to warn when the validation fails. That a decision you make in how to apply the validation, not the author of the shape.

> The name of the property signals "shape", the resource is called "shape resource".. 

Agree that the property name is misleading - it might be best to change it to something else like "constraint" or "validation". Also there's nowhere in this document that the words "shape resource" are used?

> but when the client gets there, they are directed to another resource which is the actual shape?

The purpose of this auxiliary resource is to illustrate one or more validation constraints set on that resource. It's not limited only to the shape itself, but also other parameters related to how that validation should be evaluated / applied.

Reply:
>The purpose of this auxiliary resource is to illustrate one or more validation constraints set on that resource. It's not limited only to the shape itself, but also other parameters related to how that validation should be evaluated / applied.

That's fine but I don't see a reason why that's not all (including all constraints as well as any other related information) be stated in the auxiliary shape resource.

>A resource representation whose shape is constrained by a given ShEx schema that links to an auxiliary resource defining that schema. 

Sounds closer to what I'm saying than what you're saying but perhaps I'm misinterpreting.

In any case, I can't draw the underlying rationale from the text or arguments. I'm suspecting two reasons for having an intermediary (auxiliary) resource:

* the lifecycle of the shape (constraints) information shouldn't depend on the lifecycle of the primary resource, so the auxiliary resource hop is a way to let the shapes live on. On deleting primary resource, auxiliary resource can get deleted without touching the shapes (which happen to be elsewhere and possibly used by other resources).

* there are multiple approaches to shape validation (SHACL/ShEx) so instead of picking one to associate resources with, we have the intermediary auxiliary resource to point at them.

The first seems reasonable.

The second less so - after all, no particular reason why there can't be specific link relations point at shacl or shex resources. On that note, at least one of these things needs to be required in order to have minimum level of interop.